### ISSUE TITLE
Comma 3: tools/cabana is being built on device, and causing build errors due to missing QT files

### ISSUE THREAD

**Issue by michaelhonan** [commaai/openpilot#31688]
### Describe the bug

Building openpilot (master branch) on my comma device results in this:

```
scons: Reading SConscript files ...                                                                                                                                                                                                                                                                                         scons: done reading SConscript files.                                                                                                                                                                                                                                                                                       scons: Building targets ...                                                                                                                                                                                                                                                                                                 arm-none-eabi-gcc -o panda/board/bootstub-panda.o -c -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld -Ipanda/board -Ipanda -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/bootstub.c                                                                                                                                                                                   arm-none-eabi-gcc -o panda/board/bootstub-panda_h7.o -c -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA -DENABLE_SPI -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld -Ipanda/board -Ipanda -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/bootstub.c                                                                                                                                                                    arm-none-eabi-gcc -o panda/board/jungle/bootstub-panda_jungle.o -c -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld -Ipanda/board/jungle -Ipanda/board -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/bootstub.c                                                                                                                                                 arm-none-eabi-gcc -o panda/board/jungle/bootstub-panda_jungle_h7.o -c -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld -Ipanda/board/jungle -Ipanda/board -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/bootstub.c                                                                                                                                               arm-none-eabi-gcc -o panda/board/jungle/main-panda_jungle_h7.o -c -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld -Ipanda/board/jungle -Ipanda/board -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/jungle/main.c                                                                                                                                                arm-none-eabi-gcc -o panda/board/jungle/main-panda_jungle.o -c -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld -Ipanda/board/jungle -Ipanda/board -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/jungle/main.c                                                                                                                                                  arm-none-eabi-gcc -o panda/board/jungle/obj/bootstub.panda_jungle_h7.elf -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld panda/board/jungle/obj/startup_panda_jungle_h7.o panda/board/jungle/rsa-panda_jungle_h7.o panda/board/jungle/sha-panda_jungle_h7.o panda/board/jungle/bootstub-panda_jungle_h7.o                                                                                               arm-none-eabi-objcopy -O binary panda/board/jungle/obj/bootstub.panda_jungle_h7.elf panda/board/jungle/obj/bootstub.panda_jungle_h7.bin                                                                                                                                                                                     arm-none-eabi-gcc -o panda/board/main-panda.o -c -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld -Ipanda/board -Ipanda -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/main.c                                                                                                                                                                                           arm-none-eabi-gcc -o panda/board/jungle/obj/bootstub.panda_jungle.elf -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld panda/board/jungle/obj/startup_panda_jungle.o panda/board/jungle/rsa-panda_jungle.o panda/board/jungle/sha-panda_jungle.o panda/board/jungle/bootstub-panda_jungle.o                                                                                                             arm-none-eabi-objcopy -O binary panda/board/jungle/obj/bootstub.panda_jungle.elf panda/board/jungle/obj/bootstub.panda_jungle.bin                                                                                                                                                                                           arm-none-eabi-gcc -o panda/board/main-panda_h7.o -c -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA -DENABLE_SPI -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld -Ipanda/board -Ipanda -Ipanda -Ipanda/board -Ipanda/board/stm32fx/inc -Ipanda/board/stm32h7/inc panda/board/main.c                                                                                                                                                                            arm-none-eabi-gcc -o panda/board/obj/bootstub.panda.elf -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld panda/board/obj/startup_panda.o panda/board/rsa-panda.o panda/board/sha-panda.o panda/board/bootstub-panda.o                                                                                                                                                                                          arm-none-eabi-objcopy -O binary panda/board/obj/bootstub.panda.elf panda/board/obj/bootstub.panda.bin                                                                                                                                                                                                                       arm-none-eabi-gcc -o panda/board/obj/bootstub.panda_h7.elf -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA -DENABLE_SPI -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld panda/board/obj/startup_panda_h7.o panda/board/rsa-panda_h7.o panda/board/sha-panda_h7.o panda/board/bootstub-panda_h7.o                                                                                                                                                               gcc -o panda/tests/libpanda/panda.os -c -nostdlib -fno-builtin -std=gnu11 -Wfatal-errors -Wno-pointer-to-int-cast -fPIC -Ipanda/tests/libpanda -Ipanda/board panda/tests/libpanda/panda.c                                                                                                                                   arm-none-eabi-objcopy -O binary panda/board/obj/bootstub.panda_h7.elf panda/board/obj/bootstub.panda_h7.bin                                                                                                                                                                                                                 clang++ -o selfdrive/boardd/tests/test_boardd_usbprotocol -Wl,--as-needed -Wl,--no-undefined -Wl,-rpath=/data/openpilot/third_party/acados/larch64/lib -Wl,-rpath=/usr/local/lib selfdrive/boardd/tests/test_boardd_usbprotocol.o -L/usr/local/lib -L/usr/lib -L/system/vendor/lib64 -Lthird_party/acados/larch64/lib -Lthird_party/snpe/larch64 -Lthird_party/libyuv/larch64/lib -L/usr/lib/aarch64-linux-gnu -Lcereal -Lthird_party -Lopendbc/can -Lselfdrive/boardd -Lcommon -Lrednose/helpers selfdrive/boardd/libpanda.a -lusb-1.0 common/libcommon.a -ljson11 cereal/libcereal.a cereal/libmessaging.a -lpthread -lzmq -lcapnp -lkj               arm-none-eabi-gcc -o panda/board/jungle/obj/panda_jungle.elf -Wl,--section-start,.isr_vector=0x8004000 -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld panda/board/jungle/obj/startup_panda_jungle.o panda/board/jungle/main-panda_jungle.o                                                                                                                                                            gcc -o panda/tests/libpanda/libpanda.so -shared panda/tests/libpanda/panda.os                                                                                                                                                                                                                                               arm-none-eabi-objcopy -O binary panda/board/jungle/obj/panda_jungle.elf panda/board/jungle/obj/panda_jungle.bin                                                                                                                                                                                                             SETLEN=1 panda/crypto/sign.py panda/board/jungle/obj/panda_jungle.bin panda/board/jungle/obj/panda_jungle.bin.signed panda/certs/debug                                                                                                                                                                                      /usr/bin/ld: skipping incompatible /usr/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/usr/bin/ld: skipping incompatible /usr/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
signing 53020 bytes
hash: a0e26c61939a7d7faf66c9fe2e1ef8e66397951b
clang++ -o tools/cabana/cabana.o -c -std=c++1z -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QUICK_LIB -DQT_QUICKWIDGETS_LIB -DQT_QML_LIB -DQT_CORE_LIB -DQT_MESSAGELOGCONTEXT -Wno-deprecated-declarations -Wno-deprecated-declarations -DOPENDBC_FILE_PATH='"/data/openpilot/opendbc"' -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -DENABLE_MAPS -Ithird_party/opencl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/maplibre-native-qt/include -Ithird_party/maplibre-native-qt/larch64/include -I/usr/include -I/usr/include/aarch64-linux-gnu/qt5 -I/usr/include/aarch64-linux-gnu/qt5/QtGui/5.12.8/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtCore -I/usr/include/aarch64-linux-gnu/qt5/QtNetwork -I/usr/include/aarch64-linux-gnu/qt5/QtConcurrent -I/usr/include/aarch64-linux-gnu/qt5/QtMultimedia -I/usr/include/aarch64-linux-gnu/qt5/QtQuick -I/usr/include/aarch64-linux-gnu/qt5/QtQml -I/usr/include/aarch64-linux-gnu/qt5/QtQuickWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtLocation -I/usr/include/aarch64-linux-gnu/qt5/QtPositioning -I/usr/include/aarch64-linux-gnu/qt5/QtDBus -I/usr/include/aarch64-linux-gnu/qt5/QtXml tools/cabana/cabana.cc
arm-none-eabi-gcc -o panda/board/obj/panda.elf -Wl,--section-start,.isr_vector=0x8004000 -mcpu=cortex-m4 -mhard-float -DSTM32F4 -DSTM32F413xx -mfpu=fpv4-sp-d16 -fsingle-precision-constant -Os -g -DPANDA -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32fx/stm32f4_flash.ld panda/board/obj/startup_panda.o panda/board/main-panda.o
arm-none-eabi-gcc -o panda/board/obj/panda_h7.elf -Wl,--section-start,.isr_vector=0x8020000 -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA -DENABLE_SPI -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld panda/board/obj/startup_panda_h7.o panda/board/main-panda_h7.o
arm-none-eabi-gcc -o panda/board/jungle/obj/panda_jungle_h7.elf -Wl,--section-start,.isr_vector=0x8020000 -mcpu=cortex-m7 -mhard-float -DSTM32H7 -DSTM32H725xx -mfpu=fpv5-d16 -fsingle-precision-constant -Os -g -DPANDA_JUNGLE -DALLOW_DEBUG -Wall -Wextra -Wstrict-prototypes -Werror -mlittle-endian -mthumb -nostdlib -fno-builtin -std=gnu11 -fmax-errors=1 -Tpanda/board/stm32h7/stm32h7x5_flash.ld panda/board/jungle/obj/startup_panda_jungle_h7.o panda/board/jungle/main-panda_jungle_h7.o
clang++ -o tools/cabana/streams/socketcanstream.o -c -std=c++1z -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QUICK_LIB -DQT_QUICKWIDGETS_LIB -DQT_QML_LIB -DQT_CORE_LIB -DQT_MESSAGELOGCONTEXT -Wno-deprecated-declarations -Wno-deprecated-declarations -DOPENDBC_FILE_PATH='"/data/openpilot/opendbc"' -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -DENABLE_MAPS -Ithird_party/opencl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/maplibre-native-qt/include -Ithird_party/maplibre-native-qt/larch64/include -I/usr/include -I/usr/include/aarch64-linux-gnu/qt5 -I/usr/include/aarch64-linux-gnu/qt5/QtGui/5.12.8/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtCore -I/usr/include/aarch64-linux-gnu/qt5/QtNetwork -I/usr/include/aarch64-linux-gnu/qt5/QtConcurrent -I/usr/include/aarch64-linux-gnu/qt5/QtMultimedia -I/usr/include/aarch64-linux-gnu/qt5/QtQuick -I/usr/include/aarch64-linux-gnu/qt5/QtQml -I/usr/include/aarch64-linux-gnu/qt5/QtQuickWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtLocation -I/usr/include/aarch64-linux-gnu/qt5/QtPositioning -I/usr/include/aarch64-linux-gnu/qt5/QtDBus -I/usr/include/aarch64-linux-gnu/qt5/QtXml tools/cabana/streams/socketcanstream.cc
arm-none-eabi-objcopy -O binary panda/board/jungle/obj/panda_jungle_h7.elf panda/board/jungle/obj/panda_jungle_h7.bin
arm-none-eabi-objcopy -O binary panda/board/obj/panda_h7.elf panda/board/obj/panda_h7.bin
arm-none-eabi-objcopy -O binary panda/board/obj/panda.elf panda/board/obj/panda.bin
SETLEN=1 panda/crypto/sign.py panda/board/obj/panda_h7.bin panda/board/obj/panda_h7.bin.signed panda/certs/debug
SETLEN=1 panda/crypto/sign.py panda/board/jungle/obj/panda_jungle_h7.bin panda/board/jungle/obj/panda_jungle_h7.bin.signed panda/certs/debug
SETLEN=1 panda/crypto/sign.py panda/board/obj/panda.bin panda/board/obj/panda.bin.signed panda/certs/debug
clang++ -o tools/cabana/chart/chartswidget.o -c -std=c++1z -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QUICK_LIB -DQT_QUICKWIDGETS_LIB -DQT_QML_LIB -DQT_CORE_LIB -DQT_MESSAGELOGCONTEXT -Wno-deprecated-declarations -Wno-deprecated-declarations -DOPENDBC_FILE_PATH='"/data/openpilot/opendbc"' -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -DENABLE_MAPS -Ithird_party/opencl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/maplibre-native-qt/include -Ithird_party/maplibre-native-qt/larch64/include -I/usr/include -I/usr/include/aarch64-linux-gnu/qt5 -I/usr/include/aarch64-linux-gnu/qt5/QtGui/5.12.8/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtCore -I/usr/include/aarch64-linux-gnu/qt5/QtNetwork -I/usr/include/aarch64-linux-gnu/qt5/QtConcurrent -I/usr/include/aarch64-linux-gnu/qt5/QtMultimedia -I/usr/include/aarch64-linux-gnu/qt5/QtQuick -I/usr/include/aarch64-linux-gnu/qt5/QtQml -I/usr/include/aarch64-linux-gnu/qt5/QtQuickWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtLocation -I/usr/include/aarch64-linux-gnu/qt5/QtPositioning -I/usr/include/aarch64-linux-gnu/qt5/QtDBus -I/usr/include/aarch64-linux-gnu/qt5/QtXml tools/cabana/chart/chartswidget.cc
signing 61880 bytes
hash: 4f242886ef71c137762d3f890fd2dcd0dff49d0c
In file included from tools/cabana/streams/socketcanstream.cc:1:
./tools/cabana/streams/socketcanstream.h:5:10: fatal error: 'QtSerialBus/QCanBus' file not found
#include <QtSerialBus/QCanBus>
         ^~~~~~~~~~~~~~~~~~~~~
signing 68816 bytes
hash: 192d68ff080889c85b8d957a9cebb6f076519d68
signing 60584 bytes
hash: 789b3deccc4e46f87ee95773cf7f4de458d8d029
clang++ -o tools/cabana/chart/chart.o -c -std=c++1z -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QUICK_LIB -DQT_QUICKWIDGETS_LIB -DQT_QML_LIB -DQT_CORE_LIB -DQT_MESSAGELOGCONTEXT -Wno-deprecated-declarations -Wno-deprecated-declarations -DOPENDBC_FILE_PATH='"/data/openpilot/opendbc"' -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -DENABLE_MAPS -Ithird_party/opencl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/maplibre-native-qt/include -Ithird_party/maplibre-native-qt/larch64/include -I/usr/include -I/usr/include/aarch64-linux-gnu/qt5 -I/usr/include/aarch64-linux-gnu/qt5/QtGui/5.12.8/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtCore -I/usr/include/aarch64-linux-gnu/qt5/QtNetwork -I/usr/include/aarch64-linux-gnu/qt5/QtConcurrent -I/usr/include/aarch64-linux-gnu/qt5/QtMultimedia -I/usr/include/aarch64-linux-gnu/qt5/QtQuick -I/usr/include/aarch64-linux-gnu/qt5/QtQml -I/usr/include/aarch64-linux-gnu/qt5/QtQuickWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtLocation -I/usr/include/aarch64-linux-gnu/qt5/QtPositioning -I/usr/include/aarch64-linux-gnu/qt5/QtDBus -I/usr/include/aarch64-linux-gnu/qt5/QtXml tools/cabana/chart/chart.cc
clang++ -o tools/cabana/chart/tiplabel.o -c -std=c++1z -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QUICK_LIB -DQT_QUICKWIDGETS_LIB -DQT_QML_LIB -DQT_CORE_LIB -DQT_MESSAGELOGCONTEXT -Wno-deprecated-declarations -Wno-deprecated-declarations -DOPENDBC_FILE_PATH='"/data/openpilot/opendbc"' -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -DENABLE_MAPS -Ithird_party/opencl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/maplibre-native-qt/include -Ithird_party/maplibre-native-qt/larch64/include -I/usr/include -I/usr/include/aarch64-linux-gnu/qt5 -I/usr/include/aarch64-linux-gnu/qt5/QtGui/5.12.8/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtCore -I/usr/include/aarch64-linux-gnu/qt5/QtNetwork -I/usr/include/aarch64-linux-gnu/qt5/QtConcurrent -I/usr/include/aarch64-linux-gnu/qt5/QtMultimedia -I/usr/include/aarch64-linux-gnu/qt5/QtQuick -I/usr/include/aarch64-linux-gnu/qt5/QtQml -I/usr/include/aarch64-linux-gnu/qt5/QtQuickWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtLocation -I/usr/include/aarch64-linux-gnu/qt5/QtPositioning -I/usr/include/aarch64-linux-gnu/qt5/QtDBus -I/usr/include/aarch64-linux-gnu/qt5/QtXml tools/cabana/chart/tiplabel.cc
clang++ -o tools/cabana/chart/sparkline.o -c -std=c++1z -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QUICK_LIB -DQT_QUICKWIDGETS_LIB -DQT_QML_LIB -DQT_CORE_LIB -DQT_MESSAGELOGCONTEXT -Wno-deprecated-declarations -Wno-deprecated-declarations -DOPENDBC_FILE_PATH='"/data/openpilot/opendbc"' -g -fPIC -O2 -Wunused -Werror -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DQCOM2 -mcpu=cortex-a57 -DSWAGLOG="\"common/swaglog.h\"" -DENABLE_MAPS -Ithird_party/opencl/include -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -Ithird_party/maplibre-native-qt/include -Ithird_party/maplibre-native-qt/larch64/include -I/usr/include -I/usr/include/aarch64-linux-gnu/qt5 -I/usr/include/aarch64-linux-gnu/qt5/QtGui/5.12.8/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtGui -I/usr/include/aarch64-linux-gnu/qt5/QtCore -I/usr/include/aarch64-linux-gnu/qt5/QtNetwork -I/usr/include/aarch64-linux-gnu/qt5/QtConcurrent -I/usr/include/aarch64-linux-gnu/qt5/QtMultimedia -I/usr/include/aarch64-linux-gnu/qt5/QtQuick -I/usr/include/aarch64-linux-gnu/qt5/QtQml -I/usr/include/aarch64-linux-gnu/qt5/QtQuickWidgets -I/usr/include/aarch64-linux-gnu/qt5/QtLocation -I/usr/include/aarch64-linux-gnu/qt5/QtPositioning -I/usr/include/aarch64-linux-gnu/qt5/QtDBus -I/usr/include/aarch64-linux-gnu/qt5/QtXml tools/cabana/chart/sparkline.cc
1 error generated.
scons: *** [tools/cabana/streams/socketcanstream.o] Error 1
In file included from tools/cabana/chart/chart.cc:1:
./tools/cabana/chart/chart.h:10:10: fatal error: 'QtCharts/QChartView' file not found
#include <QtCharts/QChartView>
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
scons: *** [tools/cabana/chart/chart.o] Error 1
In file included from tools/cabana/chart/chartswidget.cc:12:
./tools/cabana/chart/chart.h:10:10: fatal error: 'QtCharts/QChartView' file not found
#include <QtCharts/QChartView>
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
scons: *** [tools/cabana/chart/chartswidget.o] Error 1
In file included from tools/cabana/cabana.cc:10:
./tools/cabana/streams/socketcanstream.h:5:10: fatal error: 'QtSerialBus/QCanBus' file not found
#include <QtSerialBus/QCanBus>
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
scons: *** [tools/cabana/cabana.o] Error 1
scons: building terminated because of errors.
```

### Provide a route where the issue occurs

N/A

### openpilot version

1728355498d1e5307e7e96324b9232692d62eab6

### Additional info

comma three device, dongle id: dc30ef2ad294a732

**Comment by michaelhonan** [comment]
Looks like Cabana is being built on the device because `extras` is being included in the build options.

Knocking out the build has allowed OP to build

![image](https://github.com/commaai/openpilot/assets/12524358/a6a160c4-abe5-44a5-9492-b8f12432ab05)

**Comment by adeebshihadeh** [comment]
This should fix: https://github.com/commaai/openpilot/pull/31736. I only tested replay, and normal on-device builds only build what's necessary for driving.

**Comment by michaelhonan** [comment]
> What version of AGNOS are you running? `cat /VERSION`

9.7

**Comment by michaelhonan** [comment]
> This should fix: #31736. I only tested replay, and normal on-device builds only build what's necessary for driving.

That fixed it! Cheers.
